### PR TITLE
Datetime: Added more tests to the files in the datetime folder

### PR DESCRIPTION
**Pull request by RoxanaAnamariaTurc**
**What is this feature?**

Added more tests to files in the datetime folder.

**Why do we need this feature?**

To ensure the files have as many tests as possible to cover different scenarios.

**Who is this feature for?**

Everyone.

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: packages/grafana-data/src/datetime/datemath.test.ts [hunk 1]
```diff
@@ -183,6 +183,26 @@ describe('DateMath', () => {
...
+      const result = dateMath.isMathString(date);
+      expect(result).toBe(false);
+    });
+
+    it('should return false when invalid date text', () => {
+      expect(dateMath.isMathString('2 + 2')).toBe(false);
+    });
+
+    it('should return false if no text is passed', () => {
+      expect(dateMath.isMathString('')).toBe(false);
```
Comment (tskarhed) [inline packages/grafana-data/src/datetime/datemath.test.ts:202]: Does this account for whitespace too? If so, maybe add another assertion for a space` `

File: packages/grafana-data/src/datetime/formatter.test.ts [hunk 2]
```diff
@@ -72,5 +72,91 @@ describe('dateTimeFormat', () => {
...
+    });
+    it('should format with default formatting in correct time zone', () => {
+      const options = { timeZone: 'America/Chicago' };
+      expect(dateTimeFormat(1587126975779, options)).toBe('2020-04-17 07:36:15');
+    });
+    it('should format with default formatting in correct time zone', () => {
+      const options = { timeZone: 'America/Denver' };
+      expect(dateTimeFormat(1587126975779, options)).toBe('2020-04-17 06:36:15');
+    });
+  });
```
Comment (tskarhed) [inline packages/grafana-data/src/datetime/formatter.test.ts:103]: This is a great use case for a parametrized test! https://jestjs.io/docs/api#testeachtablename-fn-timeout

You  just create a table with the input and expected output, which reduces quite a lot of code duplication here.

File: packages/grafana-data/src/datetime/formatter.test.ts [hunk 3]
```diff
@@ -72,5 +72,91 @@ describe('dateTimeFormat', () => {
...
+  describe('DateTimeFormatISO', () => {
+    it('should format with correct ISO formatting', () => {
+      const options = { timeZone: 'Europe/Stockholm', format: 'YYYY-MM-DDTHH:mm:ss.SSSZ' };
+      expect(dateTimeFormat(1587126975779, options)).toBe('2020-04-17T14:36:15.779+02:00');
+    });
+    it('should format with correct ISO formatting', () => {
+      const options = { timeZone: 'America/New_York', format: 'YYYY-MM-DDTHH:mm:ss.SSSZ' };
+      expect(dateTimeFormat(1587126975779, options)).toBe('2020-04-17T08:36:15.779-04:00');
+    });
+    it('should format with correct ISO formatting', () => {
```
Comment (tskarhed) [inline packages/grafana-data/src/datetime/formatter.test.ts:114]: Update the names of the tests here so they differ a bit

**Review by mckn — APPROVED** [review]
LGTM! Amazing work adding this 🙌🏻

File: packages/grafana-data/src/datetime/datemath.test.ts (hunk 1 above)
Comment (RoxanaAnamariaTurc) [reply packages/grafana-data/src/datetime/datemath.test.ts:202]: Thank you for feedback , I have made the updates required based on the suggestions.

File: packages/grafana-data/src/datetime/formatter.test.ts [hunk 4]
```diff
@@ -66,18 +66,17 @@ describe('dateTimeFormat', () => {
...
-    it.each(cases)('should format with default formatting in correct time zone', (dateInUtc, options, expected) => {
+    it.each([
+      [1587126975779, { timeZone: 'Africa/Djibouti' }, '2020-04-17 15:36:15'],
+      [1587126975779, { timeZone: 'Europe/London' }, '2020-04-17 13:36:15'],
+      [1587126975779, { timeZone: 'Europe/Berlin' }, '2020-04-17 14:36:15'],
+      [1587126975779, { timeZone: 'Europe/Moscow' }, '2020-04-17 15:36:15'],
+      [1587126975779, { timeZone: 'Europe/Madrid' }, '2020-04-17 14:36:15'],
+      [1587126975779, { timeZone: 'America/New_York' }, '2020-04-17 08:36:15'],
+      [1587126975779, { timeZone: 'America/Chicago' }, '2020-04-17 07:36:15'],
+      [1587126975779, { timeZone: 'America/Denver' }, '2020-04-17 06:36:15'],
```
Comment (tskarhed) [inline packages/grafana-data/src/datetime/formatter.test.ts:78]: You don't have to put the timestamp here, since it is the same for all cases

**Review by tskarhed — APPROVED** [review]

File: packages/grafana-data/src/datetime/formatter.test.ts (hunk 4 above)
Comment (RoxanaAnamariaTurc) [reply packages/grafana-data/src/datetime/formatter.test.ts]: Thank you :)

**Pull request merged**